peachjam/templatetags/peachjam.py

```diff
@@ -37,3 +38,32 @@ def get_proper_elided_page_range(p, number, on_each_side=3, on_ends=2):
     return paginator.get_elided_page_range(
         number=number, on_each_side=on_each_side, on_ends=on_ends
     )
+
+
+@register.simple_tag
+def query_string(*args, **kwargs):
+    """
+    Combines dictionaries of query parameters and individual query parameters
+    and builds an encoded URL query string from the result.
+    """
+    query_dict = QueryDict(mutable=True)
+
+    for a in args:
+        query_dict.update(a)
+
+    remove_keys = []
+
+    for k, v in kwargs.items():
+        if v is None:
+            remove_keys.append(k)
+        elif isinstance(v, list):
+            query_dict.setlist(k, v)
+        else:
+            query_dict[k] = v
+
+    for k in remove_keys:
+        if k in query_dict:
+            del query_dict[k]
+
+    qs = query_dict.urlencode()
+    return f"?{qs}" if qs else ""
```

Comment:
you have "?" both here and in the template. I'd say keep it in the template and drop it here, because this helper doesn't know if it's building the entire querystring or not.